### PR TITLE
[RDY] Fix flamethrower sound effects and add railgun sound effect

### LINKED ISSUE #18976: The flamethrower goes bang!
FX entries haven't been updated since the split between gasoline and napalm.

### PR DESCRIPTION
**Pull request by Leland**
#18802 introduced the "flammable" ammo group used for flamethrowers, but the sound effect used for flamethrowers was limited to gasoline ammo. Updated that here, which fixes #18976.

Additionally, there were sound effects for one specific *kind* of railgun, but not the rest. I changed it so that any gun using rail rifle ammo would produce that sound.

Finally, I removed a note from the H&K G80 Railgun that described the gun as "silent while still deadly." The gun absolutely makes noise – perhaps even too much, see #20898.

PS. is it weird to anyone else that some sound effects are capitalized? I can make 'em all lowercase if so.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
If the flamethrower is loaded with gasoline using these changes, does it display the bang message when fired?

**Comment by BorkBorkGoesTheCode** [comment]
Would it be possible to attach sound effects to sound effect flags?

Sound effect flag discussion: http://smf.cataclysmdda.com/index.php?topic=14471.0

File: src/ranged.cpp [hunk 1]
```diff
@@ -1862,13 +1862,16 @@ item::sound_data item::gun_noise( bool const burst ) const
 
     noise = std::max( noise, 0 );
 
+    // Grenade launchers
     if( ammo_type() == ammotype( "40mm" ) ) {
         return { 8, _( "Thunk!" ) };
 
-    } else if( typeId() == "hk_g80") {
+    // Railguns
+    } else if( ammotype() == "12mm" || ammotype( "metal_rail" ) ) {
```
Comment (Toothspit) [inline src/ranged.cpp]: [All broke and junk](http://i.imgur.com/0CSIoEB.png), needs to be

`    } else if( ammo_type() == ammotype( "12mm" ) || ammo_type() == ammotype( "metal_rail" ) ) {`

Did this locally, got [the correct behavior](http://i.imgur.com/0Z51MrW.png).

Comment (Toothspit) [inline src/ranged.cpp]: [All broke and junk](http://i.imgur.com/0CSIoEB.png), needs to be

`    } else if( ammo_type() == ammotype( "12mm" ) || ammo_type() == ammotype( "metal_rail" ) ) {`

Did this locally, got [the correct behavior](http://i.imgur.com/0Z51MrW.png).

Comment (Leland) [inline src/ranged.cpp]: Oh man, thank you. Going back through this and flamethrowers weren't producing the right sound effect anymore. *d'oh*

Comment (Leland) [inline src/ranged.cpp]: Fixed.

**Comment by Leland** [comment]
@BorkBorkGoesTheCode first, yes: good for napalm and gasoline alike.
[Naplam](https://cloud.githubusercontent.com/assets/714017/25880758/4e3fc6cc-3507-11e7-9454-f53b15079ccb.png) and [firing](https://cloud.githubusercontent.com/assets/714017/25880760/4e4218b4-3507-11e7-9150-326cb4dce0c0.png) | [Gasoline](https://cloud.githubusercontent.com/assets/714017/25880761/4e42be9a-3507-11e7-979b-7f8467290fb5.png) and [firing](https://cloud.githubusercontent.com/assets/714017/25880759/4e3fe65c-3507-11e7-8eb2-40a0ea3f01a1.png)

Second, did some digging around in the code – so, when a weapon is fired, the sound effect file that is triggered is comprised of a [simple list of sound types](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/SOUNDPACKS.md#json-format) and [the ID of the weapon](https://github.com/CleverRaven/Cataclysm-DDA/blob/867fcd335463ab185f07e8974558bb5d16268f4f/src/sounds.cpp#L605) being used. 

If I'm reading your suggestion correctly, we actually have more flexibility in our current approach than if we were to hook it into `play_variant_sound()`, which only supports singular weapon IDs. E.g. if we wanted "tz-CRACKck" to appear for every railgun, and did so through `play_variant_sound()`, we'd have to watch for `play_variant_sound( "fire_gun", "heavy_rail_rifle")` and `play_variant_sound( "fire_gun", "rebar_rifle")`... etc.

File: src/ranged.cpp [hunk 2]
```diff
@@ -1862,13 +1862,16 @@ item::sound_data item::gun_noise( bool const burst ) const
 
     noise = std::max( noise, 0 );
 
+    // Grenade launchers
     if( ammo_type() == ammotype( "40mm" ) ) {
         return { 8, _( "Thunk!" ) };
 
-    } else if( typeId() == "hk_g80") {
+    // Railguns
+    } else if( ammotype() == "12mm" || ammotype() == "metal_rail" ) {
```
Comment (Toothspit) [inline src/ranged.cpp]: Gives a [kerblam](http://i.imgur.com/NRWOPCb.png) now. This if will never evaluate to true, `ammo_type()` is the function you're looking for and [its type is ammotype](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/item.cpp#L4255), can't compare with a string. This:
`} else if( ammo_type() == ammotype( "12mm" ) || ammo_type() == ammotype( "metal_rail" ) ) {`
gives the [cracklies](http://i.imgur.com/sPP9fYJ.png).

**Comment by Leland** [comment]
This should be ready to go.

File: src/ranged.cpp (hunk 2 above)
Comment (Leland) [inline src/ranged.cpp]: Wow, no more late night commits (...he writes at 11pm). Thanks!

**Pull request merged**